### PR TITLE
Show a diff of modified packages between bakes

### PR DESCRIPTION
**Pull request by philmcmahon**
## What does this change?
This change makes use of google's [diff-match-patch](https://github.com/google/diff-match-patch) library to generate a diff of the package lists of the current bake and the previous bake. This results in a more useful indication of what changed in the current bake.

## How to test
Perform at least 2 bakes on a recipe with some change between them. You should see the diff show up on the bake page. Locally you can experiment with [this recipe](http://localhost:9000/recipes/node-is-theft/bakes/3)

## How can we measure success?
Next time someone has an issue they suspect is caused by a change in an AMI it should be more obvious to them what has changed. 

## Images

Screenshot 2021-02-02 at 10 19 32

### CODE REVIEW

**Comment by jorgeazevedo** [comment]
There's an @ missing from the error message in [app/views/showBake.scala.html](https://github.com/guardian/amigo/pull/535/files#diff-689d377ff6d2700d9fe1e6e6a3a22610899a39be617d48e292ab22086d13b805R56) and there's a layout issue with the placement of the message.

AMIgo_and_amigo_—_-bash_—_207×38_and_AMIgo

I think that message paragraph needs to be in the same column as the "This data comes from the OS default package manager[...]" paragraph. I tried it locally and it seems to fix the issue

AMIgo

**Comment by jorgeazevedo** [comment]
Is this the second message here also intended to be in a well? http://localhost:9000/recipes/df-test/bakes/2

AMIgo

**Comment by philmcmahon** [comment]
> Is this the second message here also intended to be in a well? http://localhost:9000/recipes/df-test/bakes/2

Yep good spot!

**Review by jorgeazevedo — APPROVED** [review]
I think this is in a good state for us to deploy and get some feedback! Maybe we can open a ticket to write down some ideas for future tweaks

File: public/stylesheets/main.css [hunk 1]
```diff
@@ -13,14 +13,27 @@ body { padding-top: 70px; }
...
+    overflow-y: scroll;
+    max-height: 400px; /* This gets updated to match the screen size using JS */
+    color: #DCDCDC; /* iTerm2's default theme */
 }
 
-.bake-log {
-    color: #DCDCDC; /* iTerm2's default theme */
+.package-diff-insert {
+    color:rgb(239, 246, 252);
+    background: rgb(10, 108, 54);
```
Comment (akash1810) [inline public/stylesheets/main.css:25]:
```suggestion
    color: #EFF6FC;
    background: #0A6C36;
```

A small change to keep consistent colour declaration styles within the repo.

File: public/stylesheets/main.css [hunk 2]
```diff
@@ -13,14 +13,27 @@ body { padding-top: 70px; }
...
+    text-decoration: underline;
+}
+
+.package-diff-delete {
+    color: rgb(239, 246, 252);
+    background: rgb(144, 37, 42);
```
Comment (akash1810) [inline public/stylesheets/main.css:31]:
```suggestion
    color: #EFF6FC;
    background: #90252A;
```

A small change to keep consistent colour declaration styles within the repo.

**Pull request merged**